.github/workflows/publish.yml

```diff
@@ -0,0 +1,31 @@
+# Copyright 2023 Shukant Pal
+# SPDX-License-Identifier: Apache-2.0
+name: Publish to PyPI
+
+on:
+  workflow_dispatch:
+
+jobs:
+  pypi:
+    runs-on: ubuntu-latest
+    steps:
```

Comment:
Add

```
# Hack to workaround "on:" not supporting AND condition.
- name: Exit if not on main branch
  if: endsWith(github.ref, 'main') == false
  run: exit 0
```

Reply:
Done. I changed it to `exit 1` so it fails instead of succeeds.

Reply:
`exit 0` is correct. The idea is to make it as if the build was never run. Please change it back.

Reply:
`exit 0` will cause the step to exit successfully and it will not stop the GitHub action from continuing.

Reply:
I think we can split it into 2 jobs. The first one skips a step if not on main branch, and the second one checks if that step ran in the first job using the `if` syntax.

https://stackoverflow.com/questions/60589373/how-to-force-to-exit-in-github-actions-step#61832535

Reply:
Let's try the 2 jobs idea. However step outputs have been deprecated since that SO answer was written. You now have to write environment variables into the file `$GITHUB_ENV` and look for those variables in subsequent steps or jobs.

Reply:
Hi @MarkCallow, after taking a fresh look; it seems like a single `if` condition on the branch at the job-level should suffice. Please see my latest change.

(The 2 step workflow made sense if you can’t directly resolve the “if” true/false in without running a command)

Reply:
Sorry, I didn’t delete the old code. I pinged you on the new line added (it’s above).